StereoKitC/tools/file_picker.cpp

```diff
@@ -435,30 +461,114 @@ void file_picker_update() {
 			if (fp_active == nullptr) ui_push_enabled(false);
 			if (ui_button("Open")) { snprintf(fp_filename, sizeof(fp_filename), "%s%c%s", fp_path.folder, platform_path_separator_c, fp_active); fp_call = true; fp_call_status = true; }
 			ui_sameline();
+			ui_push_enabled(true);
+			if (ui_button("ListMode")) {
+				fp_list_mode = !fp_list_mode;
+			}
+			ui_sameline();
+			if (fp_active == nullptr) ui_push_enabled(false);
 			ui_label(fp_active?fp_active:"None selected...");
 			if (fp_active == nullptr) ui_pop_enabled();
 		} break;
 		}
 
 		ui_hseparator();
 
+		sprite_t spr_folder = sprite_find(ui_default_id_spr_arrow_right);
```

Comment:
This should be done once and cached rather than referenced every step. This is also a reference counted asset that should be released when finished!